In [2]:
from mygcca_t import *
%load_ext autoreload
%autoreload 2 

Using Theano backend.


## gcca for average precision

In [30]:
dg = data_generate()
dg.en_es_fr(800, False)

gcca = GeneralizedCCA(ds=dg, m_rank=50)
gcca.solve_u()

In [31]:
corr_array = gcca.cal_correlation(rank = True)

In [32]:
corr_array

[1.0,
 0.9999999999999999,
 0.9999999999999998,
 1.0,
 0.9999999999999998,
 0.9999999999999997,
 1.0,
 1.0,
 1.0,
 1.0,
 0.9999999999999999,
 0.9999999999999998,
 0.9999999999999999,
 0.9999999999999999,
 1.0,
 1.0,
 0.9999999999999999,
 0.9999999999999999,
 0.9999999999999999,
 0.9999999999999999,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 1.0,
 0.9999999999999998,
 0.9999999999999994,
 0.9999999999999999,
 1.0,
 1.0,
 1.0,
 0.9999999999999998,
 0.9999999999999996,
 1.0,
 0.9999999999999999,
 0.9999999999999999,
 0.9999999999999998,
 0.9999999999999998,
 1.0,
 0.9999999999999998,
 1.0,
 0.9999999999999999,
 1.0,
 0.9999999999999999,
 1.0,
 1.0,
 1.0,
 1.0]

In [33]:
gcca.cal_average_precision(gcca.list_projection)

1.0

In [34]:
v1_test_p, v2_test_p = gcca.transform(dg.test_data[0].T, dg.test_data[1].T)

In [35]:
print (v1_test_p.shape)

(720, 50)


In [36]:
gcca.cal_average_precision([v1_test_p, v2_test_p])

0.9697145061728409

## cca for average precision

In [3]:
from sklearn.cross_decomposition import CCA

In [6]:
dg = data_generate()
dg.en_es_fr(800, False)

cca = CCA(n_components=50)

In [7]:
print(dg.train_data[0].shape)
print(dg.train_data[1].shape)

(11649, 160)
(19558, 160)


In [ ]:
cca.fit(dg.train_data[0].T, dg.train_data[1].T)

In [ ]:
X_c, Y_c = cca.transform(dg.train_data[0].T, dg.train_data[1].T)

In [ ]:
gcca.cal_average_precision([X_c, Y_c])

In [ ]:
v1_test_p, v2_test_p = cca.transform(dg.test_data[0].T, dg.test_data[1].T)

In [ ]:
gcca.cal_average_precision([v1_test_p, v2_test_p])

## deep cca

## conclusion

降成50维,800个样本，训练集:测试集 = 1:4 | 训练集 | 测试集 
----|------|---- |----
deep cca | 99%  | 98% 
gcca | 100%  | 97%
cca  |  95%   | 68%

降成50维,1000个样本 | 训练集 | 测试集 
----|------|---- |----
deep cca | 93%  | 99% 
gcca | 100%  | 72%
cca  |  很差，很慢   | 很差很慢